lib/manager/terraform/lockfile/hash.ts

```diff
@@ -3,168 +3,137 @@ import extract from 'extract-zip';
 import pMap from 'p-map';
 import { join } from 'upath';
 import { TerraformProviderDatasource } from '../../../datasource/terraform-provider';
-import type {
-  TerraformBuild,
-  VersionDetailResponse,
-} from '../../../datasource/terraform-provider/types';
+import type { TerraformBuild } from '../../../datasource/terraform-provider/types';
 import { logger } from '../../../logger';
 import * as packageCache from '../../../util/cache/package';
+import { cache } from '../../../util/cache/package/decorator';
 import * as fs from '../../../util/fs';
 import { ensureCacheDir } from '../../../util/fs';
 import { Http } from '../../../util/http';
-import { repositoryRegex } from './util';
-
-const http = new Http(TerraformProviderDatasource.id);
-const hashCacheTTL = 10080; // in seconds == 1 week
-
-export async function hashFiles(files: string[]): Promise<string> {
-  const rootHash = crypto.createHash('sha256');
-
-  for (const file of files) {
-    // build for every file a line looking like "aaaaaaaaaaaaaaa  file.txt\n"
-    const hash = crypto.createHash('sha256');
-
-    // a sha256sum displayed as lowercase hex string to root hash
-    const fileBuffer = await fs.readFile(file);
-    hash.update(fileBuffer);
-    hash.end();
-    const data = hash.read();
-    rootHash.update(data.toString('hex'));
-
-    // add double space, the filename and a new line char
-    rootHash.update('  ');
-    const fileName = file.replace(/^.*[\\/]/, '');
-    rootHash.update(fileName);
-    rootHash.update('\n');
-  }
 
-  rootHash.end();
-  const rootData = rootHash.read();
-  const result: string = rootData.toString('base64');
-  return result;
-}
+export class TerraformProviderHash {
+  static http = new Http(TerraformProviderDatasource.id);
 
-export async function hashOfZipContent(
-  zipFilePath: string,
-  extractPath: string
-): Promise<string> {
-  await extract(zipFilePath, { dir: extractPath });
-  const files = await fs.readdir(extractPath);
-  // the h1 hashing algorithms requires that the files are sorted by filename
-  const sortedFiles = files.sort((a, b) => a.localeCompare(b));
-  const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+  static terraformDatasource = new TerraformProviderDatasource();
 
-  const result = await hashFiles(filesWithPath);
+  static hashCacheTTL = 10080; // in seconds == 1 week
```

Comment:
This should be minutes, as our cache layer uses minutes

https://github.com/renovatebot/renovate/blob/d0434d5df1516fdb2ef29917df41564153f5330c/lib/util/cache/package/index.ts#L28-L28